fake.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package testing
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+// FakeCallArgs holds all the args passed in a call.
+type FakeCallArgs map[string]interface{}
```

Comment:
Why do you need the name at all? Args as []interface{} for a given call should be sufficient, as well as Results []interface{}, rather than Errors. See comments below as well.